src/Autofac/Core/Resolving/ResolveOperation.cs

```diff
@@ -54,5 +72,232 @@ public object Execute(ResolveRequest request)
+            }
+
+            if (_ended)
+            {
+                throw new ObjectDisposedException(ResolveOperationResources.TemporaryContextDisposed, innerException: null);
+            }
+
+            // Create a new request context.
+            var requestContext = new ResolveRequestContext(this, request, currentOperationScope, DiagnosticSource);
+
+            // Raise our request-beginning event.
+            var handler = ResolveRequestBeginning;
+            handler?.Invoke(this, new ResolveRequestBeginningEventArgs(requestContext));
+
+            RequestDepth++;
+
+            // Track the last active request and scope in the call stack.
+            IResolveRequestContext? lastActiveRequest = ActiveRequestContext;
+            var lastScope = CurrentScope;
+
+            ActiveRequestContext = requestContext;
+            CurrentScope = currentOperationScope;
+
+            try
+            {
+                // Same basic flow in if/else, but doing a one-time check for diagnostics
+                // and choosing the "diagnostics enabled" version vs. the more common
+                // "no diagnostics enabled" path: hot-path optimization.
+                if (DiagnosticSource.IsEnabled())
+                {
+                    DiagnosticSource.RequestStart(this, requestContext);
+                    InvokePipeline(request, requestContext);
+                    DiagnosticSource.RequestSuccess(this, requestContext);
+                }
+                else
+                {
+                    InvokePipeline(request, requestContext);
+                }
+            }
+            catch (Exception ex)
+            {
+                if (DiagnosticSource.IsEnabled())
+                {
+                    DiagnosticSource.RequestFailure(this, requestContext, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                ActiveRequestContext = lastActiveRequest;
+                CurrentScope = lastScope;
+
+                // Raise the appropriate completion events.
+                if (RequestStack.Count == 0)
+                {
+                    CompleteRequests();
+                }
+
+                RequestDepth--;
+            }
+
+            // InvokePipeline throws if the instance is null but
+            // analyzers don't pick that up and get mad.
+            return requestContext.Instance!;
+        }
+
+        /// <summary>
+        /// Enter a new dependency chain block where subsequent requests inside the operation are allowed to repeat
+        /// registrations from before the block.
+        /// </summary>
+        /// <returns>A disposable that should be disposed to exit the block.</returns>
+        public IDisposable EnterNewDependencyDetectionBlock() => RequestStack.EnterSegment();
+
+        /// <inheritdoc/>
+        public SegmentedStack<IResolveRequestContext> RequestStack { get; } = new SegmentedStack<IResolveRequestContext>();
```

Comment:
Small nit, could we move this property to before any public methods (i.e. with the other properties)?

Reply:
Addressed with latest commit